src/basic/format-util.h

```diff
@@ -61,7 +61,15 @@ typedef enum {
         FORMAT_IFNAME_IFINDEX_WITH_PERCENT = (1 << 1) | FORMAT_IFNAME_IFINDEX,
 } FormatIfnameFlag;
 
-char *format_ifname_full(int ifindex, char buf[static IF_NAMESIZE + 1], FormatIfnameFlag flag);
+int format_ifname_full_with_negative_errno(int ifindex, char buf[static IF_NAMESIZE + 1], FormatIfnameFlag flag);
+static inline int format_ifname_with_negative_errno(int ifindex, char buf[static IF_NAMESIZE + 1]) {
+        return format_ifname_full_with_negative_errno(ifindex, buf, 0);
+}
+static inline char *format_ifname_full(int ifindex, char buf[static IF_NAMESIZE + 1], FormatIfnameFlag flag) {
```

Comment:
hmm, is it really worth keeping the version that doesn't return an errno? I mean, in a later commit you add FORMAT_IFNAME() as a macro, and maybe it suffices eating up the errno there, and we don#t need a funciton wrapper that hides the errno then anymore? i.e. are there many calls where it's nicer to call the errno-less function than the errno-less macro wrapper?